Mage/src/main/java/mage/abilities/effects/common/ExileTargetCardCopyAndCastEffect.java

```diff
@@ -0,0 +1,59 @@
+package mage.abilities.effects.common;
+
+import mage.ApprovingObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+
+public class ExileTargetCardCopyAndCastEffect extends OneShotEffect {
+
+    private final boolean optional;
+
+    public ExileTargetCardCopyAndCastEffect() {
+        this(true);
+    }
+
+    public ExileTargetCardCopyAndCastEffect(boolean optional) {
```

Comment:
I don't see any cards with this effect that aren't optional, and I'd be quite surprised if one was printed that wasn't optional. But in general I'm happy to include parameters so I guess it's fine.

Reply:
I added this specifically because of [[Spelltwine]]. The wording of "play something if able" is used every now and then, so I figured it would be a safe idea to include.

Reply:
I stand corrected. Sounds good.